server/models/Checklist.js

```diff
@@ -0,0 +1,116 @@
+const mongoose = require("mongoose");
+const BasePlugin = require("./BasePlugin");
+
+const checklistItemSchema = new mongoose.Schema({
+  checklistItem: {
+    type: String,
+  },
+  isChecked: {
+    type: Boolean,
+    default: false,
+  },
+});
+
+const ChecklistPluginSchema = BasePlugin.discriminator(
+  "ChecklistPlugin",
+  new mongoose.Schema(
+    {
+      checklist: [checklistItemSchema],
+    },
+    { timestamps: true }
+  )
+);
+
+ChecklistPluginSchema.methods.get = async ({ params }) => {
+  const { checklistItemId } = params;
+  if (!checklistItemId) {
+    throw new Error("checklistItemId is not defined");
+  }
+  const data = await this.model("ChecklistPlugin").findById(checklistItemId);
+  if (data) {
+    return { status: 200, data };
+  }
+  throw new Error("something went wrong");
```

Comment:
I would rather have this respond with a new status code or 500. 

`{ response: 400, message }`

Reply:
Please update where applicable in the action handlers.